GoogleForCreators/web-stories-wp
# Initial plumbing for text sets

## Summary

This PR adds the plumbing to get "Text Sets" working in the Editor.

## Relevant Technical Choices

I decided to follow a Templates approach to storing and loading our Text Sets.  The process is as follows:

1.) Create a new story using the Editor.
2.) Referencing the text sets in this [design doc](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=837%3A1962), create one text set (using the tools of the Editor) per page of the story (i.e. a 10 page story should have only 10 text sets defined).  The design doc has each text set in rows of 10.  I suggest one story of 10 pages be created for each row of text sets in the doc.  In the end we should have 9 stories that encompass _all_ text sets.
3.) Once a story has been completed with _one_ text set per page, hold `shift+option+command+j` in the Editor to open up the debug window.
4.) Copy the contents of the debug window to `**/textSets/raw/<name of row from design doc>.json`

From the text set story JSON file, our code will load the story, apply any needed migrations, and then parse out the text sets from each page (it will assume every non-background element on a page is part of _one_ text set... hence the one-text-set-per-page rule).

Right now, I'm dynamically parsing and loading the text sets directly from the story JSON files.  However, once we have dozens of text sets to parse and load, we can easily write a script that'll parse and generate text set JSON files for us to commit.  Then we'll be able to load the files directly (as needed based on filters) without additional processing.

## Things this PR doesn't do
Plenty of stuff left to do, this is just the beginning.
- Complete the rest of the text sets story files.
- Write script to auto-generate individual text set files from source story files.
- Add drag and drop functionality to text sets.
- Update "Text Set" panel to look more like mocks (with scrolling support).
- Add ability to filter text sets and load only those that are needed.
- Implement virtualization on the text set results so not all 100+ text sets get loaded into memory
- Tests and more... 

## User-facing changes

None.  Hidden behind the text set flag.

## Testing Instructions

1.) Click to "Enable text sets" in the "Experiments" page.
2.) Go to a new or existing story.
3.) Click on the "T" to open up the text tab in the editor.
4.) Notice the "Text Sets" area with 2 text sets available to use.
5.) Click on either text set and watch as it magically appears at the top/left corner of the active page.  The text set should appear properly formatted.

Fixes #4194
Fixes #3965 

## Screenshots

Text tab with "Text Sets" at the bottom:
<img src="https://user-images.githubusercontent.com/40646372/91223496-93c7f880-e6d5-11ea-9405-461d190df1a5.png" height="300">

Adding a text set to the current page:
![text_set_sample](https://user-images.githubusercontent.com/40646372/91223882-28caf180-e6d6-11ea-956a-4ba591d63d1e.gif)
